### ISSUE TITLE
commitPrefixes not working

### ISSUE THREAD

**Issue by max-two** [jesseduffield/lazygit#2215]
**Describe the bug**
I'm pretty certain this is user error rather than an actual bug, but I didn't know where else to ask.  I have just installed lazygit and wanted to add a commit prefix but I cannot get it to work and I'm unsure how to debug this.

I have added the exact config from the docs and tested the regex in regexr.com against my branch name (trevordavis/sc-138414/migrate-selectinterviewers-to-rktq) and it matches correctly.

**To Reproduce**

1. Add configuration from https://github.com/jesseduffield/lazygit/blob/master/docs/Config.md#predefined-commit-message-prefix to config file
2. Make a commit on a branch that should match the regex pattern (trevordavis/sc-138414/migrate-selectinterviewers-to-rktq in my case)
3. See that commit does not have prefix in commit message

**Expected behavior**
I would expect the prefix to be in the commit message

**Screenshots**
My config (the nerd font icons do show up, so I'm confident the config is being read from):
![image](https://user-images.githubusercontent.com/21112335/195187308-98245180-8efa-4400-931c-65c9b82c0092.png)

I did try to run in debug mode and saw this error log when running a commit.  Not entirely sure what to make of it though:
![image](https://user-images.githubusercontent.com/21112335/195188123-81741efb-2333-4291-a859-9b738f0e8810.png)

**Version info:**
_Run `lazygit --version` and paste the result here_
commit=367b0d331836c90c015bf0c45f88612f3d94d08a, build date=2022-07-20T09:27:56Z, build source=binaryRelease, version=0.35, os=darwin, arch=amd64

_Run `git --version` and paste the result here_
git version 2.37.3

**Comment by HarrisonFretwell** [comment]
Also tried copying the posted example and couldn't get it to work?

**Comment by longlhh90** [comment]
Yesterday, when I updated lazygit to the newest version, my commitPrefixes is not working anymore. Actually, it's not working when I start lazygit and start committing. If I commit it as WIP and cancel and recommit again prefix will show. 

It was working perfectly fine in old version (I couldn't remember which one).

**Comment by jesseduffield** [comment]
This is fixed now on master (by https://github.com/jesseduffield/lazygit/pull/2612). I'll push a new patch for this soon

**Comment by zeromask1337** [comment]
I got same bug, but it behaves differently.
![image](https://github.com/jesseduffield/lazygit/assets/22420621/1bb97520-5e02-4b5b-97bc-dd08717ff542)

I have my config like this 
```yaml
gui:
  language: 'en'
git:
  commitPrefixes:
    black_ore_frontend:
      pattern: "/([a-zA-Z0-9]+-?\\d+)/"
      replace: '[$1] some text lol'

```

And for some reason my commit message only has branch name, no brackets or other text. Tried different methods, such as commenting parts of config. Pretty confident that its either my missunderstanding of config, or bug with reading `replace` property.
